### ISSUE TITLE
`local:exec`, `local:docker`, `cluster:k8s`: Prometheus+pushgateway healthchecks/fix

### ISSUE THREAD

**Issue by raulk** [ipfs/testground#660]
We have introduced compulsory infrastructure elements (Pushgateway) but have not updated the healthchecks on runners.

Right now, plans run with `local:exec` will always print this error:

```
  15.5428s    MESSAGE << instance  92 >> error during periodic metric push: %!w(*url.Error=&{Post http://prometheus-pushgateway:9091/metrics/job@base64/dGVzdGdyb3VuZC9wbGFu/ContainerName/Rauls-MBP-2.lan/TestPlan/benchmarks/TestCase/all/TestRun/cf31cf7a98fe/TestGroupID/single/TestCaseSeq/%00/TestCommit//TestTag/ 0xc000142d70})
```

This is because the runenv is scheduling a periodic Prometheus push assuming that there's a pushgateway listening at host `prometheus-pushgateway`, which is not the case.

This problem will also happen with all other runners, unless the user takes manual steps. Let's add the healthchecks + fix code to install/run Prometheus and Pushgateway automatically!

## Runners to update (in order of priority)

- [x] `local:docker`
- [ ] `cluster:k8s`
- [x] `local:exec`

**Comment by Robmat05** [comment]
Added to sprint as P1 bug

**Comment by coryschwartz** [comment]
The change in https://github.com/ipfs/testground/pull/573 is not that complicated, but it is old enough that dealing with the merge conflicts is more difficult than starting over, so I'm closing that one and I'll open another one fresh.

**Comment by coryschwartz** [comment]
Regarding step 1, which is to be able to actually run prometheus and pushgateway on local docker,
The first PR is here, which provides an "EnsureImage" 

https://github.com/ipfs/testground/pull/681/
That is there to provide pre-configured prometheus image which is configured to scrape the pushgateway.

This is different than the method I was using in https://github.com/ipfs/testground/pull/573 of bind mounting the config.

In another PR, I'm going to add a the pre-configured image files to infra/docker.

**Comment by coryschwartz** [comment]
The local exec checks PR will close this issue once it is merged.

**Comment by Robmat05** [comment]
Not sure why this is in Review/QA as it is an issue and needs no review. Is this closed or should stay in progress?

**Comment by coryschwartz** [comment]
I expect this will be completed tomorrow, I think @raulk will review and merge https://github.com/ipfs/testground/pull/734 and then this will be completed in the processs

**Comment by raulk** [comment]
@coryschwartz the cluster:k8s runner is not performing healthchecks on prometheus and friends, so this issue can't be closed.

**Comment by raulk** [comment]
Maybe drop `cluster:k8s` from the scope of this issue, and open up another issue for that.
